src/hooks/useNftRetry.tsx

```diff
@@ -0,0 +1,166 @@
+import { useCallback, useContext, useMemo, useState } from 'react';
+import { ContentIsLoadedEvent, ShimmerActionContext } from 'contexts/shimmer/ShimmerContext';
+import { usePromisifiedMutation } from 'hooks/usePromisifiedMutation';
+import { graphql } from 'relay-runtime';
+import { useToastActions } from 'contexts/toast/ToastContext';
+import { useReportError } from 'contexts/errorReporting/ErrorReportingContext';
+import { CouldNotRenderNftError } from 'errors/CouldNotRenderNftError';
+import { Primitive } from 'relay-runtime/lib/store/RelayStoreTypes';
+import { useNftRetryMutation } from '../../__generated__/useNftRetryMutation.graphql';
+
+type useNftRetryArgs = {
+  tokenId: string;
+};
+
+type useNftRetryResult = {
+  retryKey: number;
+  isFailed: boolean;
+  refreshMetadata: () => void;
+  refreshingMetadata: boolean;
+  handleNftLoaded: ContentIsLoadedEvent;
+  handleNftError: (error: Error) => void;
+};
+
+export function useNftRetry({ tokenId }: useNftRetryArgs): useNftRetryResult {
+  const reportError = useReportError();
+  const { pushToast } = useToastActions();
+  const shimmerContext = useContext(ShimmerActionContext);
+
+  const [isFailed, setIsFailed] = useState(false);
+  const [retryKey, setRetryKey] = useState(0);
+  const [refreshed, setRefreshed] = useState(false);
+  const [refreshingMetadata, setRefreshingMetadata] = useState(false);
+
+  const handleNftLoaded = useCallback(
+    (event?: any) => {
+      shimmerContext?.setContentIsLoaded(event);
+      setIsFailed(false);
+    },
+    [shimmerContext]
+  );
+
+  const handleNftError = useCallback(
+    (error: Error) => {
+      // Give up and show the failure state
+      shimmerContext?.setContentIsLoaded(event);
+      setIsFailed(true);
+
+      // If the user refreshed the metadata and there was another failure,
+      // we'll show them a new toast telling them things failed to load,
+      // and we're looking into the issue asap.
+      if (refreshed) {
+        pushToast({
+          message:
+            'This piece has failed to load. This issue has been reported to the Gallery team.',
+          autoClose: true,
+        });
+      }
+
+      const commonTags = {
+        tokenId,
+        alreadyRefreshed: refreshed,
+      };
+
+      if (error instanceof CouldNotRenderNftError) {
+        reportError(error.message, {
+          tags: { ...commonTags, ...error.metadata },
+        });
+      } else {
+        reportError('Could not load nft', {
+          tags: commonTags,
+        });
+      }
```

Comment:
TODO: come up with a process to routinely check on these errors

Reply:
I added a prefix to report error so they should be easy to create a search for